File: src/main/groovy/net/zomis/brainf/model/ast/TokenInfo.java
package net.zomis.brainf.model.ast;

public class TokenInfo {

    int position;
    int length;
    int line;
    int column;
    String file;

    public TokenInfo(TokenInfo tokenInfo) {
        this.position = tokenInfo.position;
        this.length = tokenInfo.length;
        this.line = tokenInfo.line;
        this.file = tokenInfo.file;
    }

    public TokenInfo() {

    }

    public void reset() {
        position += length;
        length = 1;
    }

    public int getLength() {
        return length;
    }

    public int getLine() {
        return line;
    }

    public int getPosition() {
        return position;
    }

    public String getFile() {
        return file;
    }

    public int getColumn() {
        return column;
    }

    @Override
    public String toString() {
        return "TokenInfo{" +
                "position=" + position +
                ", length=" + length +
                ", line=" + line +
                ", column=" + column +
                ", file='" + file + '\'' +
                '}';
    }
}

File: brainduck-kotlin/src/commonMain/kotlin/net/zomis/brainf/model/ast/TokenInfo.kt
package net.zomis.brainf.model.ast;

data class TokenInfo(
    var position: Int,
    var length: Int,
    var line: Int,
    val column: Int,
    val file: String,
) {
    // TODO: Old code did not copy column - was that a bug or a feature?

    fun reset() {
        position += length;
        length = 1;
    }

    override fun toString(): String {
        return "TokenInfo{" +
                "position=" + position +
                ", length=" + length +
                ", line=" + line +
                ", column=" + column +
                ", file='" + file + '\'' +
                '}';
    }

    companion object {
        fun createNew(): TokenInfo = TokenInfo(
            position = 0,
            length = 0,
            line = 0,
            column = 0,
            file = "",
        )
    }
}
